lib/runAll.js

```diff
@@ -68,75 +68,89 @@ module.exports = async function runAll(
 `)
   }
 
-  const tasks = generateTasks({ config, cwd, gitDir, files, relative }).map(task => ({
-    title: `Running tasks for ${task.pattern}`,
-    task: async () =>
-      new Listr(
-        await makeCmdTasks({
-          commands: task.commands,
-          files: task.fileList,
-          gitDir,
-          logger,
-          shell
-        }),
-        {
+  const tasks = generateTasks({ config, cwd, gitDir, files, relative })
+
+  // lint-staged 10 will automatically add modifications to index
+  // Warn user when their command includes `git add`
+  let hasDeprecatedGitAdd = false
+
+  const listrTasks = tasks.map(task => {
+    const subTasks = makeCmdTasks({ commands: task.commands, files: task.fileList, gitDir, shell })
+
+    if (subTasks.some(subTask => subTask.command.includes('git add'))) {
+      hasDeprecatedGitAdd = true
+    }
+
+    return {
+      title: `Running tasks for ${task.pattern}`,
+      task: async () =>
+        new Listr(subTasks, {
           // In sub-tasks we don't want to run concurrently
           // and we want to abort on errors
           dateFormat: false,
           concurrent: false,
           exitOnError: true
+        }),
+      skip: () => {
+        if (task.fileList.length === 0) {
+          return `No staged files match ${task.pattern}`
         }
-      ),
-    skip: () => {
-      if (task.fileList.length === 0) {
-        return `No staged files match ${task.pattern}`
+        return false
       }
-      return false
     }
-  }))
+  })
 
-  const listrOptions = {
-    dateFormat: false,
-    renderer: (quiet && 'silent') || (debug && 'verbose') || 'update'
+  if (hasDeprecatedGitAdd) {
+    logger.warn(`${symbols.warning} ${chalk.yellow(
+      `Detected a task using \`git add\`. Lint-staged version 10 will automatically add any task modifications to the git index, and you should remove this command.`
```

Comment:
Let's remove `version 10` since it will be a true for further versions as well. I'd rewrite like this:

``Some of your tasks use \`git add\` command. Please remove it from the config since all modifications made by tasks will be automatically added to the git commit index.``

Maybe we should add a link to the readme section there too? Thoughts?